ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -1183,7 +1189,12 @@ void Matmul::validate(
                 // tensor in1
                 TT_FATAL(input_tensor_b.memory_config().memory_layout == TensorMemoryLayout::WIDTH_SHARDED, "Error");
             } else if constexpr (std::is_same_v<ProgramConfigType, MatmulMultiCoreReuseMultiCastProgramConfig>) {
+                TT_FATAL(program_config.per_core_M % program_config.out_block_h == 0, "Error");
+                TT_FATAL(program_config.per_core_N % program_config.out_block_w == 0, "Error");
+                TT_FATAL(program_config.out_block_h % program_config.out_subblock_h == 0, "Error");
+                TT_FATAL(program_config.out_block_w % program_config.out_subblock_w == 0, "Error");
```

Comment:
These should print out something like "Error: incompatible values {} and {}"

Reply:
done